src/EtherFiNode.sol

```diff
@@ -483,54 +485,55 @@ contract EtherFiNode is IEtherFiNode {
+        return Address.functionCall(eigenPod, _data);
     }
 
-    // As an optimization, it skips the call to 'etherFiNodesManager' back again to retrieve the target address
-    function forwardCall(address to, bytes memory data) external onlyEtherFiNodeManagerContract returns (bytes memory) {
-        _verifyForwardCall(to, data);
-        return Address.functionCall(to, data);
+    function forwardCall(address _to, bytes calldata _data) external onlyEtherFiNodeManagerContract returns (bytes memory) {
+        _verifyForwardCall(_to, _data);
+        return Address.functionCall(_to, _data);
     }
-    
-    //--------------------------------------------------------------------------------------
-    //-------------------------------  INTERNAL FUNCTIONS  ---------------------------------
-    //--------------------------------------------------------------------------------------
 
-    function _verifyEigenPodCall(bytes memory data) internal view {
-        bytes4 selector;
-        assembly {
-            selector := mload(add(data, 0x20))
-        }
+    function _verifyEigenPodCall(bytes calldata _data) internal view {
+
+        if (_data.length < 4) revert InvalidForwardedCall();
+        bytes4 selector = bytes4(_data[:4]);
+
+        // can add extra restrictions to specific calls here i.e. checking specific paramaters
+        if (!IEtherFiNodesManager(etherFiNodesManager).allowedForwardedEigenpodCalls(selector)) revert ForwardedCallNotAllowed();
 
         // withdrawNonBeaconChainETHBalanceWei
         if (selector == IEigenPod.withdrawNonBeaconChainETHBalanceWei.selector) {
-            require(data.length >= 36, "INVALID_DATA_LENGTH");
-            address recipient;
-            assembly {
-                recipient := mload(add(data, 0x24))
-            }
+            require(_data.length == 68, "INVALID_DATA_LENGTH");
+            address recipient = address(bytes20(_data[16:36]));
+
             // No withdrawal to any other address than the safe
             require (recipient == address(this), "INCORRECT_RECIPIENT");
         }
 
-        // recoverTokens(IERC20[], uint256[], address)
-        if (selector == IEigenPod.recoverTokens.selector) {
-            revert("NOT_ALLOWED");
-        }
     }
 
-    function _verifyForwardCall(address to, bytes memory data) internal view {
-        bytes4 selector;
-        assembly {
-            selector := mload(add(data, 0x20))
-        }
-        bool allowed = (selector != IDelegationManager.completeQueuedWithdrawal.selector && selector != IDelegationManager.completeQueuedWithdrawals.selector);
-        require (allowed, "NOT_ALLOWED");
+    function _verifyForwardCall(address _to, bytes calldata _data) internal view {
+
+        if (_data.length < 4) revert InvalidForwardedCall();
+        bytes4 selector = bytes4(_data[:4]);
+
+        if (!IEtherFiNodesManager(etherFiNodesManager).allowedForwardedExternalCalls(selector, _to)) revert ForwardedCallNotAllowed();
```

Comment:
I think we can move all these checks on the forward calls to `EtherFiNodesManager` contract. just need to make suer the forward call are made via `EtherFiNodesManager` contract by putting `onlyEtherFiNodeManagerContract` where needed